eZ/Publish/Core/Repository/Tests/Values/ContentType/FieldDefinitionCollectionTest.php

```diff
@@ -0,0 +1,318 @@
+            $this->createFieldDefinitionsWith('fieldTypeIdentifier', ['ezstring', 'ezstring', 'ezimage'])
+        );
+
+        $this->assertTrue($collection->anyOfType('ezstring'));
+        $this->assertFalse($collection->anyOfType('ezrichtext'));
+    }
+
+    public function testAnyInGroup(): void
+    {
+        $collection = new FieldDefinitionCollection(
+            $this->createFieldDefinitionsWith('fieldGroup', ['default', 'default', 'seo'])
+        );
+
+        $this->assertTrue($collection->anyInGroup('default'));
+        $this->assertFalse($collection->anyInGroup('comments'));
+    }
+
+    public function testPartition(): void
+    {
+        list($a, $b, $c) = $this->createFieldDefinitions('A', 'B', 'C');
+
+        $collection = new FieldDefinitionCollection([$a, $b, $c]);
+
+        $this->assertEquals(
+            [
+                new FieldDefinitionCollection([$a, $c]),
+                new FieldDefinitionCollection([$b]),
+            ],
+            $collection->partition($this->getIdentifierIsEqualPredicate('A', 'C'))
+        );
+
+        $this->assertEquals(
+            [
+                new FieldDefinitionCollection([$a, $b, $c]),
+                new FieldDefinitionCollection(),
+            ],
+            $collection->partition($this->getTautology())
+        );
+
+        $this->assertEquals(
+            [
+                new FieldDefinitionCollection(),
+                new FieldDefinitionCollection([$a, $b, $c]),
+            ],
+            $collection->partition($this->getContraction())
+        );
+    }
+
+    public function testToArray(): void
+    {
+        $fieldDefinitions = $this->createFieldDefinitions('A', 'B', 'C');
+
+        $collection = new FieldDefinitionCollection($fieldDefinitions);
+
+        $this->assertEquals($fieldDefinitions, $collection->toArray());
+    }
+
+    private function createFieldDefinitions(string ...$identifiers): array
+    {
+        return $this->createFieldDefinitionsWith('identifier', $identifiers);
+    }
+
+    private function createFieldDefinitionsWith(string $property, array $values): array
+    {
+        return array_map(function (string $identifier) use ($property): APIFieldDefinition {
+            return $this->createFieldDefinition($identifier, $property);
+        }, $values);
+    }
+
+    private function createFieldDefinition(string $identifier, string $property = 'identifier'): APIFieldDefinition
+    {
+        return new FieldDefinition([$property => $identifier]);
+    }
+
+    /**
+     * Returns predicate which test if field definition identifier belongs to given set.
+     */
+    public function getIdentifierIsEqualPredicate(string ...$identifiers): Closure
+    {
+        return static function (APIFieldDefinition $fieldDefinition) use ($identifiers): bool {
+            return in_array($fieldDefinition->identifier, $identifiers);
+        };
+    }
+
+    /**
+     * Returns predicate with is always true.
```

Comment:
```suggestion
     * Returns a predicate with is always true.
```
not quite following "with" in this sentence, but seems like it's intentional?